### PR TITLE
GDPR/Privacy Compliance added to inquiry form

### PR DESCRIPTION
**Pull request by CaitlynKnudsvig**
EBM sites need a privacy compliance checkbox added to the inquiry forms, so if the user selects one of the affected countries, a checkbox will appear with a custom label set in site.js (`gdprMessage`).  By default, the message will be null and the checkbox (`gdprOptIn`) will be set to false.

Additionally, a second message is being added at the bottom of the form, regardless of country, which can also be set in site.js (`privacyMessage`). 

I'm still a vue noob, so if there's a better way of doing handling this, just let me know and I'd be happy to make any changes. :) 

Screen Shot 2020-12-02 at 4 45 17 PM

Screen Shot 2020-12-02 at 4 45 02 PM

### CODE REVIEW

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:133]: To be a little more semantically clear, call this `isGdprCountry` please!

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:189]: There should be a boolean flag to turn on or off the GDPR functionality. By default, it should be off.

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:191]: This should have a default value. I'd suggest
> This website may use my contact information to communicate with me by email or telephone about other offerings that may be of interest to me.

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:212]: This value should respect the `isGdprEnabled` boolean and return false if disabled. This list of companies should be overridable -- make it the default value for a `gdprCountryCodes` property.

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:213]: This should use the `isGdprEnabled` property, not the `gdprMessage` property.

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:218]:
```suggestion
        return this.gdprCountryCodes.includes(this.country);
```

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:157]:
```suggestion
    <div v-if="privacyMessage" class="row">
```

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:224]:
```suggestion
```
You don't need a computed property to access a vue property, remove.

File: packages/marko-web-inquiry/components/form.marko
Comment (solocommand) [inline packages/marko-web-inquiry/components/form.marko:41]:
```suggestion
      gdprMessage: inquiry.gdprMessage,
      gdprEnabled: Boolean(inquiry.gdprEnabled),
      gdprCountryCodes: inquiry.gdprCountryCodes,
```

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (solocommand) [inline packages/marko-web-inquiry/browser/default-form.vue:154]: Why was `form-group` added to the submit button?

**Review by solocommand — CHANGES_REQUESTED** [review]
@CaitlynKnudsvig I think this is fine in concept, but will require some changes in order to work for all BaseCMS clients. Also, if further customizations need to be made beyond inclusion of these two fields, the Inquiry package *was* written so that you can inject your own Vue form component.

See https://github.com/base-cms/base-cms/blob/55b467ae36d4db84bf8d26350822db912057cd5c/packages/marko-web-inquiry/browser/index.js#L4 and https://github.com/base-cms-websites/endeavor-business-media/blob/master/packages/shared/browser/index.js#L22 -- you can pass your own custom component in and use that instead of the `DefaultForm` component.

File: packages/marko-web-inquiry/browser/default-form.vue
Comment (CaitlynKnudsvig) [reply packages/marko-web-inquiry/browser/default-form.vue:154]: That was just to add some spacing below the button.  I only chose the form-group class because the rest of the rows use that for spacing, so it was just to keep it spaced consistently, but I'd be happy to do it another way instead? :)

**Comment by CaitlynKnudsvig** [comment]
> @CaitlynKnudsvig I think this is fine in concept, but will require some changes in order to work for all BaseCMS clients. Also, if further customizations need to be made beyond inclusion of these two fields, the Inquiry package _was_ written so that you can inject your own Vue form component.
> 
> See
> 
> https://github.com/base-cms/base-cms/blob/55b467ae36d4db84bf8d26350822db912057cd5c/packages/marko-web-inquiry/browser/index.js#L4
> 
> and https://github.com/base-cms-websites/endeavor-business-media/blob/master/packages/shared/browser/index.js#L22 -- you can pass your own custom component in and use that instead of the `DefaultForm` component.

Crap!  TIL 😓 I guess I should have assumed there would be a way to do this in the website repo, or at least asked first lol.

**Comment by CaitlynKnudsvig** [comment]
Apologies for the delay; I'm still waiting on approval from the group, so let's not push this live just yet.  I would have created this as a draft PR if I had known it would take this long to hear back from them 😓 

In the meantime, I just pushed the requested changes up.  @solocommand let me know if I missed anything, or there's anything else you catch that I can update before this is merged 😄

**Review by solocommand — APPROVED** [review]

**Comment by CaitlynKnudsvig** [comment]
Finally got approval from the group on this, lmk if it's still good to merge, or if I need to make any updates :)

**Comment by CaitlynKnudsvig** [comment]
@solocommand - if it's not too much trouble, could we deploy this today?  I'm out for the year after Thursday and would like to get this live today in case anything needs to be changed/fixed before I go on vaca :)

**Comment by CaitlynKnudsvig** [comment]
@zarathustra323 - I think Josh might be out today?  If you have a sec, would you mind merging/deploying this today pretty please?

**Pull request merged**
